library/Vanilla/FeatureFlagHelper.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Vanilla;
+
+use Gdn;
+use Gdn_Configuration;
+use Exception;
+
+/**
+ * A helper class for gatekeeping code behind feature flags.
+ */
+class FeatureFlagHelper {
+
+    /**
+     * Is a feature enabled?
+     *
+     * @param string $feature The config-friendly name of the feature.
+     * @return bool
+     * @throws \Garden\Container\NotFoundException If unable to find the config class in the container.
+     * @throws \Garden\Container\ContainerException If unable to get the config class instance from the container.
+     */
+    public static function featureEnabled(string $feature): bool {
+        // We're going to enforce the root "Feature" namespace.
+        $configValue = Gdn::getContainer()->get(\Gdn_Configuration::class)->get("Feature.{$feature}");
```

Comment:
Should we be getting the config directly through DI in the constructor? My understanding was that we were trying to move away from needing to touch the `Gdn` object in new classes.

Reply:
The class is just a collection of static "helper" methods, similar to `ApiUtils`. That means that, ideally, it doesn't need to be instantiated (i.e. no constructor call for DI). If we were going to pass the config instance via instantiating and DI, whatever was using these methods would need to ensure the instance passed through the container too. Those classes would need their own separate calls to `Gdn::getContainer` to retrieve an instance or would need to add this helper class as a dependency on each of their individual constructors.

I can replace this with a call to `c` to avoid directly referencing the `Gdn` class.